lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py

```diff
@@ -278,17 +280,30 @@ def list_items(self):
         results = []
         for item in items:
             if self.has_tags(item.tags, self.tags):
-                results.append(self.serialize_vm(self.get_vm(item.name)))
+                results.append(self.serialize_vm(self.get_vm(self.resource_group, item.name)))
         return results
 
-    def get_vm(self, name):
+    def list_all_items(self):
+        self.log('List all items')
+        try:
+            items = self.compute_client.virtual_machines.list_all()
```

Comment:
is the response paged?

Reply:
yes, handled by the standard python iterator